crates/ide-db/src/path_transform.rs

```diff
@@ -151,45 +164,44 @@ impl<'a> PathTransform<'a> {
 }
 
 struct Ctx<'a> {
-    type_and_const_substs: FxHashMap<hir::TypeOrConstParam, ast::Type>,
+    type_and_const_substs: FxHashMap<hir::TypeOrConstParam, TypeOrConstSubst>,
     lifetime_substs: FxHashMap<LifetimeName, ast::Lifetime>,
     target_module: hir::Module,
     source_scope: &'a SemanticsScope<'a>,
 }
 
+fn preorder(item: &SyntaxNode) -> impl Iterator<Item = SyntaxNode> {
+    item.preorder().filter_map(|event| match event {
+        syntax::WalkEvent::Enter(_) => None,
+        syntax::WalkEvent::Leave(node) => Some(node),
+    })
+}
+
 impl<'a> Ctx<'a> {
     fn apply(&self, item: &SyntaxNode) {
+        for (_, subst) in &self.type_and_const_substs {
+            if subst.to_be_transformed {
+                let paths =
+                    preorder(&subst.subst.syntax()).filter_map(ast::Path::cast).collect::<Vec<_>>();
+                for path in paths {
+                    self.transform_path(path);
+                }
+            }
+        }
+
```

Comment:
this transformation takes place in `Ctx` as it relies on the `Ctx::transform_path` method. but it's a private method, so I can move it out of `Ctx` and use in `build_ctx()`, it makes sense

Reply:
Since it's in the same module `build_ctx()` can call `Ctx::transform_path()` just fine, so the easiest way might be to construct `Ctx` and call it before returning it.

Reply:
eventually I moved it to a separate function in `Ctx` which `buld_ctx()` calls